pandas/tests/indexing/test_loc.py

```diff
@@ -2700,3 +2700,13 @@ def test_loc_setitem(self, string_series):
         string_series.loc[d2] = 6
         assert string_series[d1] == 4
         assert string_series[d2] == 6
+
+    def test_loc_dict_assign(self):
+        # GH41044
+        dfo = DataFrame({"A": ["abc", "def"], "B": ["ghi", "jkl"]}, dtype="object")
+        dfs = DataFrame({"A": ["abc", "def"], "B": ["ghi", "jkl"]}, dtype="string")
+
+        dfo.loc[0, :] = {"A": "newA", "B": "newB"}
+        dfs.loc[0, :] = {"A": "newA", "B": "newB"}
+
+        tm.assert_frame_equal(dfo, dfs, check_dtype=False)
```

Comment:
Could you define expected once explicitly, this way we would not catch a change in behavior if this would change simultaneously

Reply:
Good idea! If we're going to define the expected explicitly, maybe we can parametrize the dtype possibilities so this can be expanded beyond `object` and `string`. Would that make sense?

Reply:
Yes